whois-commons/src/main/resources/nrtm_schema.sql

```diff
@@ -19,58 +19,78 @@ create table `source`
 ) ENGINE = InnoDB
   DEFAULT CHARSET = utf8;
 
-DROP TABLE IF EXISTS `version_information`;
-create table `version_information`
+DROP TABLE IF EXISTS `version`;
+create table `version`
 (
     `id`             int unsigned NOT NULL AUTO_INCREMENT,
     `source_id`      int unsigned NOT NULL,
-    `version`        int unsigned NOT NULL DEFAULT '0',
-    `session_id`     varchar(128) NOT NULL DEFAULT '',
-    `type`           varchar(128) NOT NULL DEFAULT '',
-    `last_serial_id` int          NOT NULL DEFAULT '0',
+    `version`        int unsigned NOT NULL,
+    `session_id`     varchar(128) NOT NULL,
+    `type`           varchar(128) NOT NULL,
+    `last_serial_id` int          NOT NULL,
     PRIMARY KEY (`id`),
-    CONSTRAINT `version_information__source_fk` FOREIGN KEY (`source_id`) REFERENCES `source` (`id`),
-    UNIQUE KEY `version_information__source__version_uk` (`source_id`, `version`),
-    UNIQUE KEY `version_information__session__version_uk` (`session_id`, `version`),
-    UNIQUE KEY `version_information__type__last_serial_id_uk` (`type`, `last_serial_id`)
+    CONSTRAINT `version__source__fk` FOREIGN KEY (`source_id`) REFERENCES `source` (`id`),
+    UNIQUE KEY `version__source__version__uk` (`source_id`, `version`),
+    UNIQUE KEY `version__session__version__uk` (`session_id`, `version`),
+    UNIQUE KEY `version__type__last_serial_id__uk` (`type`, `last_serial_id`)
+) ENGINE = InnoDB
+  DEFAULT CHARSET = utf8;
+
+DROP TABLE IF EXISTS `notification_file`;
+create table `notification_file`
+(
+    `id`         int unsigned    NOT NULL AUTO_INCREMENT,
+    `version_id` int unsigned    NOT NULL,
+    `payload`    longtext        NOT NULL,
+    `created`    bigint unsigned NOT NULL,
+    PRIMARY KEY (`id`),
+    CONSTRAINT `notification_file__version_id__fk` FOREIGN KEY (`version_id`) REFERENCES `version` (`id`)
 ) ENGINE = InnoDB
   DEFAULT CHARSET = utf8;
 
 DROP TABLE IF EXISTS `snapshot_file`;
 create table `snapshot_file`
 (
-    `id`         int(10) unsigned NOT NULL AUTO_INCREMENT,
-    `version_id` int(10) unsigned NOT NULL DEFAULT '0',
-    `name`       varchar(256)     NOT NULL DEFAULT '',
-    `hash`       varchar(128)     NOT NULL DEFAULT '',
-    `created`    bigint unsigned  not null default unix_timestamp(),
+    `id`         int unsigned    NOT NULL AUTO_INCREMENT,
+    `version_id` int unsigned    NOT NULL,
+    `name`       varchar(256)    NOT NULL,
+    `hash`       varchar(256)    NOT NULL,
+    `created`    bigint unsigned NOT NULL,
     PRIMARY KEY (`id`),
-    CONSTRAINT `snapshot_file__version_information_fk` FOREIGN KEY (`version_id`) REFERENCES `version_information` (`id`)
+    UNIQUE KEY `snapshot_file__version_id__uk` (`version_id`),
+    UNIQUE KEY `snapshot_file__name__uk` (`name`),
+    CONSTRAINT `snapshot_file__version_id__fk` FOREIGN KEY (`version_id`) REFERENCES `version` (`id`)
 ) ENGINE = InnoDB
   DEFAULT CHARSET = utf8;
 
 DROP TABLE IF EXISTS `delta_file`;
 create table `delta_file`
 (
```

Comment:
can we rename to delta_object? if we are using snapshot_object naming convention for saving payload?